Add support for category assets

nikola-mladenovic opened the pull request:
@lauraluiz - LMK how does this look to you, and if it makes sense we can create an issue in the commercetools/commercetools-jvm-sdk project.

lauraluiz commented on commercetools-models/src/main/java/io/sphere/sdk/categories/commands/updateactions/SetAssetCustomField.java:
```diff
@@ -0,0 +1,32 @@
+        return assetId;
+    }
+
+    public static SetAssetCustomField usingJson(final String assetId, final String name, final JsonNode value) {
```
Now I'm checking the same with products, I guess you removed it because it sounded weird. Let's see... I would just simply say `ofJsonValue`, sounds better than `ofUsingJson` or `ofJson` (which is misleading). The "using" part from the other classes forces us to change a bit the structure here, which is bad, but it's the best option I can think of and yet it's easy to understand.

nikola-mladenovic replied:
I also thought about `ofJsonValue` - it lead me to think the `final JsonNode value` is the first param, and not the `final String assetId`, but perhaps that's just me, coming from the Swifty background. Let's use `ofJsonValue`, cause I don't think we'll figure out anything better here ☺️

lauraluiz replied:
Yep. I share your view: not perfect but probably the best ;)

lauraluiz commented on commercetools-models/src/main/java/io/sphere/sdk/categories/Category.java:
```diff
@@ -173,6 +174,10 @@
     @Nullable
     CustomFields getCustom();
 
+    @Nonnull
```
Okay just saw that Michael used it here for Products... I wonder why. But Products is not auto-generated and I don't see anything else problematic here, so give it a try.

lauraluiz replied:
Anyway, leave it. We found the issue elsewhere :)

nikola-mladenovic commented on commercetools-models/src/main/java/io/sphere/sdk/categories/commands/updateactions/AddAsset.java:
```diff
@@ -0,0 +1,21 @@
+
+    public AssetDraft getAsset() {
+        return asset;
+    }
```
👍 Added a static constructor method for `AddAsset` update action.

nikola-mladenovic commented:
I've updated `CategoryBuilder.java` and added setter for `assets`: `public CategoryBuilder assets(final List<Asset> assets)`, so it can be used if needed for `Category` tests.

Now the project builds properly, but I cannot run all those integration tests because of missing project credentials.

@lauraluiz - if & what else do you think we need here before merging?

nikola-mladenovic commented:
Added tests which cover category assets, there's one small outstanding issue on the API side to be addressed for `assetCustomType` test. After that's been fixed, we could go ahead with the merge.

lauraluiz commented on commercetools-models/src/main/java/io/sphere/sdk/categories/commands/updateactions/ChangeAssetOrder.java:
```diff
@@ -0,0 +1,22 @@
+public final class ChangeAssetOrder extends UpdateActionImpl<Category> {
+    private final List<String> assetOrder;
+
+    public ChangeAssetOrder(final List<String> assetOrder) {
```
Constructor should be private and there should be a static constructor available (`of` method).

lauraluiz commented on commercetools-models/src/main/java/io/sphere/sdk/categories/commands/updateactions/RemoveAsset.java:
```diff
@@ -0,0 +1,20 @@
+public final class RemoveAsset extends UpdateActionImpl<Category> {
+    private final String assetId;
+
+    public RemoveAsset(final String assetId) {
```
Same as `ChangeAssetOrder`: constructor private and creation via static constructor.

lauraluiz commented on commercetools-models/src/main/java/io/sphere/sdk/categories/commands/updateactions/SetAssetDescription.java:
```diff
@@ -0,0 +1,35 @@
+import javax.annotation.Nullable;
+
+/**
+ * Changes the name of an asset.
```
*"Changes the description"
By the way, could you fix this same mistake from the Product's action `SetAssetDescription`? ;)

nikola-mladenovic commented:
Addressed all remarks ☺️

nikola-mladenovic commented:
@lauraluiz - CI is finally green, feel free to merge when you get a chance ☺️
